js/controllers/books.js

```diff
@@ -0,0 +1,112 @@
+/* global app */
+/* global angular */
+/*global $rootScope*/
+
+app.controller('booksCtrl', ['$scope', '$state', '$stateParams', 'bookService', 'authorService', 'enableEditService', 'uiGridConstants', function($scope, $state, $stateParams, bookService, authorService, enableEditService, uiGridConstants, $index) {
+  
+  $scope.include = function(obj) {
+     bookService.updateBooks(obj);
+   };
+   
+  $scope.authorNameDropdown = authorService.authorsList; 
+  $scope.bookTemp = bookService.booksList;
+
+  $scope.genreTypes = [
+    {id:"Murder Mystery", value:"Murder Mystery"},
+    {id:"Historical Fiction", value:"Historical Fiction"},
+    {id:"Contemporary Fiction", value:"Contemporary Fiction"} 
+  ];
+  
+  $scope.genreTypesModal = [
+    "Murder Mystery", "Historical Fiction", "Contemporary Fiction" 
+  ];
+  
+  $scope.edit = enableEditService.edit;
+  
+  $scope.bookGridOptions = {
+    
+    data: 'bookTemp',
+    enableCellSelection: true,
+    enableCellEdit: true,
+    enableCellEditOnFocus: true,
+    rowEditWaitInterval: -1,
+    saveFocus: false,
+    saveScroll: true,
+    saveGroupingExpandedStates: true,
+    enableRowSelection: true,
+    onRegisterApi: function (gridApi) {
+                   $scope.gridApi = gridApi;
+                   $scope.gridApi.edit.on.afterCellEdit($scope,function(rowEntity, colDef, newValue, oldValue){
+                         if(newValue != oldValue){
+                            $state.reload();
+                         }
+                    });
+                   },
+  
+    columnDefs: [
+
+       { field: 'id', displayName: 'ID', enableCellEdit: false, enableHiding: false
+       },
+       { field: 'name', displayName: 'Name', enableHiding: false, cellEditableCondition: $scope.edit
+       },
+       { field: 'author', displayName: 'Author', enableHiding: false, cellEditableCondition: $scope.edit,
```

Comment:
it seems like in grid editing is enabled for books but not authors - is that correct? If not, please refactor accordingly. If it is correct, please provide comments about what is going on here and why

Reply:
on the books page, the book title (name), author, and genre's ability to be edited (cellEditableCondition) are based on $scope.edit, which refers back to a boolean variable stored in enableEditService. same goes for the name and age on the authors page. It is changed to true if the user logs in as someone with edit permissions, and stays false if the user does not have edit permissions. so if you log in as admin, it would be "cellEditableCondition: true".